guildwars2/account.py

```diff
@@ -379,6 +391,10 @@ def count_upgrades(slots):
             for item in v:
                 count += get_amount_in_slot(item)
             storage_counts[k] = count
+        if "tradingpost" in doc["permissions"]:
+            storage_counts["tp delivery"] = 0
+            for item in delivery:
+                storage_counts["tp delivery"] += get_amount_in_slot(item)
```

Comment:
While this won't raise a NameError now, it's still a bit messy. Just put this whole try/except block and this block under one try/except on line 394